src/main/java/use_cases/add_edit_expenses_use_case/ExpenseUCI.java

```diff
@@ -120,14 +112,14 @@ public ExpenseOD editExpenseInMonth(ExpenseID expenseIDEdit) throws EntityExcept
 
             if (changeInRecurringInfo(expenseIDEdit)) {
                 if (expenseIDEdit.getIsRecurringExpense()) {
-                    session.addRecurExpense(expenseFactory.editMonthObject(setExpenseEditorInfo(expenseIDEdit)));
+                    session.addRecurExpense((Expense) expenseFactory.editMonthObject(setExpenseEditorInfo(expenseIDEdit)));
                     // Success edit to new recurring expense.
                     ExpenseOD expenseODSuccessEditRecurring = new ExpenseOD("You have updated all changes " +
-                            "of this new recurring expense!");
+                            "of this recurring expense!");
                     return expenseOB.success(expenseODSuccessEditRecurring);
                 }else{
                     expenseFactory.editMonthObject(setExpenseEditorInfo(expenseIDEdit));
-                    session.deleteRecurExpense(expenseIDEdit.getName());
+                    session.deleteRecurExpense(expenseIDEdit.getOldExpense());
```

Comment:
Why are you using the old name? When the Expense is edited to have the new name that should change in the recurring expense list so you should be using getName() like before.

Reply:
In this case, the condition describes when changeinRecurringInfo is true and the new input is not a recurring expense. This means deleting the recurring expense should be called upon the old expense(to remove the selected expense from recurring list). expenseIDEdit.getName is the new name that is given to the oldExpense, and won't be recorded in the recurring list.

Reply:
Hmm, but the recurring list in the session is an ArrayList of Expense objects, meaning that changing the Expense objects name to a new name would update in that list (especially since it’s not a local copy you made, it’s the one stored in the session). So then the recurring expense list SHOULD hold the Expense with the new name and not the old one (otherwise that means that our code does not work, as our Expense objects are not updating).

Reply:
Remember this is only the case when the user is changing the isRecurring from True to False, hence the behaviour should be to remove the selected/old expense from the recurring list and set other changes. What you described is the general case of editing a name (and no change in isRecurring data), which is handled by line125!

Reply:
Not quite actually lol

Reply:
We decided that getName() is correct on call.